File: CodingTestPlatform/BOJ/2_Silver/4/9012/9012.py
# 괄호
# https://www.acmicpc.net/problem/9012

import sys
input = sys.stdin.readline

cnt = int(input().rstrip())

for i in range(cnt):
    s = input().rstrip()
    stk = []
    chk = True
    for j in s:
        if j == "(":
            stk.append(1)
        elif j == ")":
            rst = stk.pop() if len(stk) > 0 else -1
            if rst != 1:
                chk = False
                break
    if chk is True and len(stk) == 0:
        print("YES")
    else:
        print("NO")

File: CodingTestPlatform/BOJ/2_Silver/4/9012/9012.js
/**
 * 괄호
 * Silver4
 * https://www.acmicpc.net/problem/9012
 */

const [T, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split("\n");

const result = input.map((bracket) => {
    const b = bracket.split("");
    let leftBracket = 0;
    for (let i = 0; i < b.length; i++) {
        b[i] === "(" ? leftBracket++ : leftBracket--;
        if (leftBracket < 0) break;
    }
    return leftBracket === 0 ? "YES" : "NO";
});

console.log(result.join("\n"));